app/models/course/assessment/answer/answer_ability.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+module Course::Assessment::Answer::AnswerAbility
+  def define_permissions
+    if user
+      allow_students_create_programming_files
+      allow_students_destroy_programming_files
+    end
+
+    super
+  end
+
+  def allow_students_create_programming_files
```

Comment:
Will need to add a check the current user against the creator of the submission. The consequence of not checking means staff can actually create and delete files.

Reply:
ok done, specs have been modified as well.

Reply:
I think you need to add in `can_update_submission`, if not we allow students to create and delete programming files after the submission has been submitted? 😅

Reply:
right, used that the last time but replaced it with `creator?`, didn't realize that replacing it would have this implication haha